File: server_integration_tests/servers/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/DefaultsServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.enums.Priority;
import dev.graphlink.javaserver.generated.inputs.ConfigInput;
import dev.graphlink.javaserver.generated.inputs.RangeInput;
import dev.graphlink.javaserver.generated.services.AckPriorityService;
import dev.graphlink.javaserver.generated.services.ConfigService;
import dev.graphlink.javaserver.generated.services.EchoPriorityService;
import dev.graphlink.javaserver.generated.services.GreetService;
import dev.graphlink.javaserver.generated.services.RangeService;
import dev.graphlink.javaserver.generated.types.Config;
import dev.graphlink.javaserver.generated.types.Range;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

/**
 * Default/initial-value resolvers: each simply reflects its effective input
 * back, so a client that omits a defaulted field/argument can assert the
 * schema default was applied end-to-end.
 */
@Service
public class DefaultsServiceImpl
        implements ConfigService, RangeService, GreetService, EchoPriorityService, AckPriorityService {

    @Override
    public Mono<Config> resolveConfig(ConfigInput input) {
        return Mono.just(Config.builder()
            .pageSize(input.getPageSize())
            .ratio(input.getRatio())
            .sort(input.getSort())
            .verbose(input.getVerbose())
            .priority(input.getPriority())
            .tags(input.getTags())
            .empties(input.getEmpties())
            .note(input.getNote())
            .build());
    }

    @Override
    public Mono<Range> resolveRange(RangeInput input) {
        return Mono.just(Range.builder().min(input.getMin()).max(input.getMax()).build());
    }

    @Override
    public Mono<String> greet(String name, Integer times) {
        return Mono.just(("Hi " + name + "!").repeat(times));
    }

    @Override
    public Mono<Priority> echoPriority(Priority level) {
        return Mono.just(level);
    }

    @Override
    public Mono<String> ackPriority(Priority level) {
        return Mono.just("OK");
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/DefaultsServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.enums.Priority
import dev.graphlink.kotlinserverblocking.generated.inputs.ConfigInput
import dev.graphlink.kotlinserverblocking.generated.inputs.RangeInput
import dev.graphlink.kotlinserverblocking.generated.services.AckPriorityService
import dev.graphlink.kotlinserverblocking.generated.services.ConfigService
import dev.graphlink.kotlinserverblocking.generated.services.EchoPriorityService
import dev.graphlink.kotlinserverblocking.generated.services.GreetService
import dev.graphlink.kotlinserverblocking.generated.services.RangeService
import dev.graphlink.kotlinserverblocking.generated.types.Config
import dev.graphlink.kotlinserverblocking.generated.types.Range
import org.springframework.stereotype.Service

/**
 * Default/initial-value resolvers: each simply reflects its effective input
 * back, so a client that omits a defaulted field/argument can assert the
 * schema default was applied end-to-end.
 */
@Service
class DefaultsServiceImpl :
    ConfigService, RangeService, GreetService, EchoPriorityService, AckPriorityService {

    override fun resolveConfig(input: ConfigInput): Config = Config(
        pageSize = input.pageSize!!,
        ratio = input.ratio!!,
        sort = input.sort!!,
        verbose = input.verbose!!,
        priority = input.priority!!,
        tags = input.tags!!,
        empties = input.empties!!,
        note = input.note,
    )

    override fun resolveRange(input: RangeInput?): Range =
        Range(min = input?.min ?: 0, max = input?.max ?: 100)

    override fun greet(name: String?, times: Int?): String =
        "Hi $name!".repeat(times ?: 1)

    override fun echoPriority(level: Priority?): Priority = level ?: Priority.HIGH

    override fun ackPriority(level: Priority): String = "OK"
}
